packages/block-library/src/navigation/is-wrapping.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Determines if the children of a wrapper element are wrapping.
+ *
+ * @param {Element} wrapper  The element to determine if its children are wrapping
+ * @param {Array}   children The children of the wrapper element
+ *
+ * @return {boolean} Whether the children of the wrapper element are wrapping.
+ */
+function areItemsWrapping(
+	wrapper,
+	children = wrapper.querySelectorAll( 'li' )
+) {
+	const wrapperDimensions = wrapper.getBoundingClientRect();
+	//we store an array with the width of each item
+	const itemsWidths = getItemWidths( children );
+	let totalWidth = 0;
+	let isWrapping = false;
+
+	//the nav block may have row-gap applied, which is not calculated in getItemWidths
+	const computedStyle = window.getComputedStyle( wrapper );
+	const rowGap = parseFloat( computedStyle.rowGap ) || 0;
+
+	for ( let i = 0, len = itemsWidths.length; i < len; i++ ) {
+		totalWidth += itemsWidths[ i ];
+		if ( rowGap > 0 && i > 0 ) {
+			totalWidth += rowGap;
+		}
+		if ( parseInt( totalWidth ) > parseInt( wrapperDimensions.width ) ) {
```

Comment:
```suggestion
		if ( parseInt( totalWidth ) > parseInt( wrapperDimensionsWidth ) ) {
```

We should read the `.width` property at the top of the function in one hit. See https://web.dev/articles/avoid-large-complex-layouts-and-layout-thrashing

Reply:
Done in 4e41eb09b4